cdap-data-fabric-tests/src/test/java/co/cask/cdap/data/stream/StreamFileSizeManagerTest.java

```diff
@@ -0,0 +1,218 @@
+        @Override
+        protected void configure() {
+          bind(MetricsCollectionService.class).to(NoOpMetricsCollectionService.class);
+
+          bind(StreamConsumerStateStoreFactory.class)
+            .to(LevelDBStreamConsumerStateStoreFactory.class).in(Singleton.class);
+          bind(StreamAdmin.class).to(LevelDBStreamFileAdmin.class).in(Singleton.class);
+          bind(StreamConsumerFactory.class).to(LevelDBStreamFileConsumerFactory.class).in(Singleton.class);
+          bind(StreamFileWriterFactory.class).to(LocationStreamFileWriterFactory.class).in(Singleton.class);
+
+          bind(HeartbeatPublisher.class).to(MockHeartbeatPublisher.class).in(Scopes.SINGLETON);
+          }
+        }));
+
+    txService = injector.getInstance(TransactionManager.class);
+    txService.startAndWait();
+    dsOpService = injector.getInstance(DatasetOpExecutor.class);
+    dsOpService.startAndWait();
+    datasetService = injector.getInstance(DatasetService.class);
+    datasetService.startAndWait();
+    streamHttpService = injector.getInstance(StreamHttpService.class);
+    streamHttpService.startAndWait();
+    heartbeatPublisher = (MockHeartbeatPublisher) injector.getInstance(HeartbeatPublisher.class);
+
+    port = streamHttpService.getBindAddress().getPort();
+
+    return injector;
+  }
+
+  public static void stopGateway(CConfiguration conf) {
+    streamHttpService.stopAndWait();
+    datasetService.stopAndWait();
+    dsOpService.stopAndWait();
+    txService.stopAndWait();
+    conf.clear();
+  }
+
+  private HttpURLConnection openURL(String location, HttpMethod method) throws IOException {
+    URL url = new URL(location);
+    HttpURLConnection urlConn = (HttpURLConnection) url.openConnection();
+    urlConn.setRequestMethod(method.getName());
+    urlConn.setRequestProperty(Constants.Gateway.API_KEY, API_KEY);
+
+    return urlConn;
+  }
+
+  @Test
+  public void streamPublishesHeartbeatTest() throws Exception {
+    // Now, create the new stream.
+    HttpURLConnection urlConn = openURL(String.format("http://%s:%d/v2/streams/test_stream1", HOSTNAME, port),
+                                        HttpMethod.PUT);
+    Assert.assertEquals(HttpResponseStatus.OK.getCode(), urlConn.getResponseCode());
+    urlConn.disconnect();
+
+    // Enqueue 10 entries
+    for (int i = 0; i < 10; ++i) {
+      urlConn = openURL(String.format("http://%s:%d/v2/streams/test_stream1", HOSTNAME, port), HttpMethod.POST);
+      urlConn.setDoOutput(true);
+      urlConn.addRequestProperty("test_stream1.header1", Integer.toString(i));
+      urlConn.getOutputStream().write(TWO_BYTES);
+      Assert.assertEquals(HttpResponseStatus.OK.getCode(), urlConn.getResponseCode());
+      urlConn.disconnect();
+    }
+
+    TimeUnit.SECONDS.sleep(Constants.Stream.HEARTBEAT_DELAY + 1);
+    StreamWriterHeartbeat heartbeat = heartbeatPublisher.getHeartbeat();
+    Assert.assertNotNull(heartbeat);
+    Assert.assertEquals(20, heartbeat.getAbsoluteDataSize());
```

Comment:
Is this 20 because there are 10 entries each of 2Bytes size each? Can you compute the expected value from the individual components?